tests/meltano/cli/test_run.py

```diff
@@ -1085,6 +1085,24 @@ def test_run_mapper_config(
             assert result.exit_code == 1
             assert "Error: Block not-a-valid-mapping-name not found" in result.stderr
 
+        # test mapper/mapping name collision detection
+        args = ["run", tap.name, "mapper-mock", target.name]
```

Comment:
Think you missed adding a mapping name that collides with mapper name. Test was still passing because it would still catch when being invoked with a mapper name, but it wasn't actually being triggered by a collision.

```suggestion
        project_add_service.add(
            PluginType.MAPPERS,
            "mapper-collision",
            inherit_from=mapper.name,
            mappings=[
                {
                    "name": "mapper-collision",
                    "config": {
                        "transformations": [
                            {
                                "field_id": "author_email1",
                                "tap_stream_name": "commits1",
                                "type": "MASK-HIDDEN",
                            }
                        ]
                    },
                }
            ],
        )

        args = ["run", tap.name, "mapper-collision", target.name]
```

Reply:
Nope, that was intentional since my added parser check is just evaluating if the `PluginType.MAPPERS` block is the same as a mapper plugin name.  Say someone installed the mapper and just tried to run the command `meltano tap-csv transform-field target-postgres` it will fail out and give the error even with no mappings config present in the `meltano.yml`.  That is at least the behavior on my machine.  I will add this as an additional check.

Reply:
👍 gotcha , two explicit ones makes sense then.